Public/Src/Utilities/KeyValueStore/KeyValueStoreAccessor.cs

```diff
@@ -638,17 +736,75 @@ public Possible<Unit> Use(Action<IBuildXLKeyValueStore> use)
                     return Unit.Void;
                 }, use);
         }
+        
+        private (bool lockdown, bool rethrow) StrictErrorPolicy(Failure<Exception> failure)
+        {
+            Contract.Requires(failure != null);
+
+            var ex = failure.Content;
+            var isUserError = true;
+            if (ex is RocksDbSharpException || ex is System.Runtime.InteropServices.SEHException)
+            {
+                // The SEHException class handles SEH (structured exception handling) errors that are thrown from 
+                // unmanaged code, but that have not been mapped to another .NET Framework exception. The SEHException
+                // class also corresponds to the HRESULT E_FAIL (0x80004005).
+                isUserError = false;
+            }
+
+            m_failureHandler?.Invoke(failure);
+            return (true, isUserError);
+        }
+
+        private (bool lockdown, bool rethrow) RelaxedErrorPolicy(Failure<Exception> failure)
+        {
+            Contract.Requires(failure != null);
+
+            var ex = failure.Content;
+
+            var isRocksDbError = false;
+            if (ex is RocksDbSharpException || ex is System.Runtime.InteropServices.SEHException)
+            {
+                // The SEHException class handles SEH (structured exception handling) errors that are thrown from 
+                // unmanaged code, but that have not been mapped to another .NET Framework exception. The SEHException
+                // class also corresponds to the HRESULT E_FAIL (0x80004005).
+                isRocksDbError = true;
+            }
+
+            if (isRocksDbError)
+            {
+                return (true, false);

```

Comment:
In strict policy above, handler is called for RocksDbErrors, but in relaxed policy, it is not. Is that intentional/desirable?

Reply:
That is intentional. In the relaxed policy, only the critical error handler is called. In this way, the failure handler doesn't have to have something like "if this is critical exception, then X"